components/SimpleTextForPath.tsx

```diff
@@ -68,22 +60,21 @@ export const SimpleTextForPath = ({
       return;
     }
 
-    if (timeoutRef.current) {
-      clearTimeout(timeoutRef.current);
-    }
-
-    onSelection();
-    setIsLongPressing(true);
-    setIsSaving(true);
-    setIsSaved(false);
-    setPressIndex(index);
-    setSavedPathVerseId(verseId);
-    onSave();
-
-    timeoutRef.current = setTimeout(() => {
+    Animated.timing(new Animated.Value(0), {
+      toValue: 1,
+      duration: 100,
+      useNativeDriver: true,
+    }).start(() => {
+      onSelection();
+      setIsLongPressing(true);
+      setIsSaving(true);
+      setIsSaved(true);
```

Comment:
What is difference between states setIsSaving and setIsSaved?

Is there any use case, where these two would have different value?

In this case, both are set to true at same time.